### ISSUE TITLE
X-LIMIT-U, X-LIMIT-D is not set by iOS client (OwnTracks)

### ISSUE THREAD

**Issue by aleohl** [LINKIWI/orion-server#2]
X-LIMIT-U and X-LIMIT-D is not provided in the request from iOS.
Need to get these from some other source, basic auth?

Headers on a request from iOS:
```
Host: <redacted>
Content-Type: application/json
User-Agent: OwnTracks/9.8.4 CFNetwork/893.14.2 Darwin/17.3.0
Connection: keep-alive
Accept: */*
Accept-Language: en-us
Content-Length: 215
Accept-Encoding: gzip, deflate
Authorization: Basic <redacted>
```

**Comment by aleohl** [comment]
Also the JSON payload differs.

Could probably use `if is set topic`, set X-LIMIT-U, X-LIMIT-D from parsed topic.

iOS:
```json
{
	"batt": 42,
	"lon": REDACTED,
	"acc": 65,
	"vac": 10,
	"topic": "owntracks\/user\/uuid",
	"lat": REDACTED,
	"t": "u",
	"conn": "w",
	"tst": 1517405128,
	"alt": 8,
	"_type": "location",
	"tid": "6p"
}
```
Android:
```json
{
	"_type": "location",
	"tid": "hw",
	"acc": 22,
	"batt": 75,
	"conn": "w",
	"lat": REDACTED,
	"lon": REDACTED,
	"tst": 1517406669
}
```

**Comment by LINKIWI** [comment]
Interesting, thanks for reporting. Unfortunately I don't have an iOS device so I am not able to test this end-to-end.

I would prefer parsing the username and device from `topic` if provided, falling back to `X-Limit-U` and `X-Limit-D` when `topic` is not available. Parsing the `Authorization` header feels a bit hacky, especially since the reverse-proxying web server may change its value.

I'm happy to accept a PR for this. Alternatively, if the iOS JSON payload you pasted above is full and complete, I can use that as a test case for patching it myself.

**Comment by LINKIWI** [comment]
Fixed in #3

**Comment by taalas** [comment]
I am seeing the same problem as stated in the linked issue from orion-docker when trying to publish from iOS, using the latest version from the repository. Is there anything else I should be aware of when using OwnTracks on iOS?
